lighthouse-core/gather/driver/navigation.js

```diff
@@ -157,4 +158,18 @@ function getNavigationWarnings(navigation) {
   return warnings;
 }
 
-module.exports = {gotoURL, getNavigationWarnings, UIStrings};
+/**
+ * @param {string|undefined} url
+ * @return {string}
+ */
+function normalizeUrl(url) {
```

Comment:
Reducing mocks is one good reason to move this out of any gathering files since there's no gathering needed for it. It's a pretty good fit for url-shim, actually (sorry for the work you already did to de-mock it, but hopefully the idea makes it even simpler :)